arangod/Aql/IResearchViewOptimizerRules.cpp

```diff
@@ -305,74 +296,168 @@ bool optimizeScoreSort(IResearchViewNode& viewNode, ExecutionPlan* plan) {
-        // FIXME (Dronplane): here we should deal with stored
-        //                    values when we will support such optimization
-        [[fallthrough]];
+        switch (astCalcNode->type) {
+          case AstNodeType::NODE_TYPE_REFERENCE:
+            // something produced by during search function replacement.
+            // e.g. it is expected to be LET sortVar = scorerVar;
+            {
+              auto sortVariable =
+                  reinterpret_cast<Variable const*>(astCalcNode->getData());
+              TRI_ASSERT(sortVariable);
+              auto const s = std::find_if(
+                  std::begin(scorers), std::end(scorers),
+                  [sortVariableId = sortVariable->id](auto const& t) noexcept {
+                    return t.var->id == sortVariableId;
+                  });
+              if (s == std::end(scorers)) {
+                return false;
+              }
+              heapSort.push_back(HeapSortElement{
+                  .postfix = "",
+                  .source = std::distance(scorers.begin(), s),
+                  .fieldNumber = std::numeric_limits<size_t>::max(),
+                  .ascending = sort.ascending});
+            }
+            break;
+          case AstNodeType::NODE_TYPE_ATTRIBUTE_ACCESS:
+            if (checkAttributeAccess(astCalcNode, viewVariable, false)) {
+              // direct access to view variable
+              latematerialized::AttributeAndField<
+                  latematerialized::IndexFieldData>
+                  af;
+              // sort.attributePath is empty so we extract the name ourselves
+              std::string name;
+              if (!nameFromAttributeAccess(name, *astCalcNode, ctx, false)) {
+                return false;
+              }
+              try {
+                TRI_ParseAttributeString(name, af.attr, false);
+              } catch (::arangodb::basics::Exception const&) {
+                return false;
+              }
+              heapSort.push_back(HeapSortElement{
+                  .postfix = "",
+                  .source = 0,
+                  .fieldNumber = std::numeric_limits<size_t>::max(),
+                  .ascending = sort.ascending});
+              attrs.push_back(std::move(af));
+              storedMaps.insert({attrs.size() - 1, heapSort.size() - 1});
+            } else {
+              // this could be stored column access but with postfix.
+              HeapSortElement element;
+              size_t usedVars{0};
+              visitReferencedVariables(
+                  *astCalcNode, [&](aql::Variable const& var) {
+                    if (++usedVars < 2) {
+                      // only one variable allowed e.g. LET x = y.foo
+                      auto column = viewNode.getSourceColumnInfo(var.id);
+                      std::string name;
+                      if (column.first !=
+                              std::numeric_limits<ptrdiff_t>::max() &&
```

Comment:
it's not enough scope...
```suggestion
                                                                                                                                                                                   std::numeric_limits<ptrdiff_t>::max() &&
```

Reply:
Could you please elaborate more?